test/parachain/tests/integration_test.rs

```diff
@@ -0,0 +1,390 @@
+		fn submit_extrinsic(extrinsic: String) -> PHash;
+	}
+
+	Chain {
+		#[rpc(method = "chain_getFinalizedHead")]
+		fn current_block_hash() -> PHash;
+
+		#[rpc(method = "chain_getHeader", positional_params)]
+		fn header(hash: PHash) -> Option<Header>;
+
+		#[rpc(method = "chain_getBlockHash", positional_params)]
+		fn block_hash(hash: Option<u64>) -> Option<PHash>;
+	}
+
+	State {
+		#[rpc(method = "state_getRuntimeVersion")]
+		fn runtime_version() -> RuntimeVersion;
+	}
+
+	System {
+		#[rpc(method = "system_networkState")]
+		fn network_state() -> Value;
+	}
+}
+
+// Adapted from
+// https://github.com/rust-lang/cargo/blob/485670b3983b52289a2f353d589c57fae2f60f82/tests/testsuite/support/mod.rs#L507
+fn target_dir() -> PathBuf {
+	env::current_exe()
+		.ok()
+		.map(|mut path| {
+			path.pop();
+			if path.ends_with("deps") {
+				path.pop();
+			}
+			path
+		})
+		.unwrap()
+}
+
+struct ChildHelper<'a> {
+	name: String,
+	child: &'a mut Child,
+}
+
+impl<'a> Drop for ChildHelper<'a> {
+	fn drop(&mut self) {
+		let name = self.name.clone();
+
+		self.terminate();
+
+		let mut stdout = String::new();
+		if let Some(reader) = self.child.stdout.as_mut() {
+			let _ = reader.read_to_string(&mut stdout);
+		}
+		eprintln!("process '{}' stdout:\n{}\n", name, stdout,);
+
+		let mut stderr = String::new();
+		if let Some(reader) = self.child.stderr.as_mut() {
+			let _ = reader.read_to_string(&mut stderr);
+		}
+		eprintln!("process '{}' stderr:\n{}\n", name, stderr,);
+	}
+}
+
+impl<'a> ChildHelper<'a> {
+	fn new(name: &str, child: &'a mut Child) -> ChildHelper<'a> {
+		ChildHelper {
+			name: name.to_string(),
+			child,
+		}
+	}
+
+	fn terminate(&mut self) {
+		match self.child.try_wait() {
+			Ok(Some(_)) => return,
+			Ok(None) => {}
+			Err(err) => {
+				eprintln!("could not wait for child process to finish: {}", err);
+				let _ = self.child.kill();
+				let _ = self.child.wait();
+				return;
+			}
+		}
+
+		#[cfg(unix)]
+		{
+			use nix::sys::signal::{kill, Signal::SIGTERM};
+			use nix::unistd::Pid;
+
+			kill(Pid::from_raw(self.child.id().try_into().unwrap()), SIGTERM).unwrap();
+
+			let mut tries = 30;
+
+			let success = loop {
+				tries -= 1;
+
+				match self.child.try_wait() {
+					Ok(Some(_)) => break true,
+					Ok(None) if tries == 0 => break false,
+					Ok(None) => thread::sleep(Duration::from_secs(1)),
+					Err(err) => {
+						eprintln!("could not wait for child process to finish: {}", err);
+						break false;
+					}
+				}
+			};
+
+			if !success {
+				let _ = self.child.kill();
+			}
+		}
+
+		#[cfg(not(unix))]
+		let _ = self.child.kill();
+
+		let _ = self.child.wait();
+	}
+}
+
+async fn wait_for_tcp<A: net::ToSocketAddrs + std::fmt::Display>(address: A) {
+	while let Err(err) = net::TcpStream::connect(&address) {
+		eprintln!("Waiting for {} to be up ({})...", address, err);
+		sleep(Duration::from_secs(2)).await;
+	}
+}
+
+#[async_std::test]
+#[ignore]
```

Comment:
It's complicated.

Originally I wanted to make the test only run with `--release`. Unfortunately there isn't a flag for that and [the common way to do this is to check the flag debug_assertions instead](https://stackoverflow.com/questions/39204908/how-to-check-release-debug-builds-using-cfg-in-rust#39205417).

But... the CI explicitly enable the `debug_assertions` with `--release` during the tests. So that doesn't work.

The closest way I found to solve this issue is [here](https://www.reddit.com/r/rust/comments/3i1nki/how_to_skip_expensive_tests_with_cargo_test/)

Do you have another idea?

Reply:
Interestingly, as of 19b7942ee, running the test in debug mode fails in only 11m30s. Presumably, 90s was spent in setup, and the other 10 minutes was spent attempting to gather blocks.

The build mode of the test suite shouldn't affect the runtime of the chains themselves, which are run via the `cargo_run` helper; I imagine that that always builds in release mode regardless. 

I think this data may support running this test unconditionally, instead of gating on release.

Reply:
Nah this made my life harder. I didn't know I had to run in release and I didn't realize it was a problem (I don't think this is obvious at all). I prefer to make it very explicit that this test needs to run in release, even adding a note or a safeguard if I could.